src/core/database/database.service.ts

```diff
@@ -218,48 +222,86 @@ export class DatabaseService {
    */
   async sgUpdateProperties<TObject extends Resource>({
     object,
-    props,
     changes,
     nodevar,
   }: {
     session: Session;
     object: TObject;
-    props: ReadonlyArray<keyof TObject & string>;
     changes: { [Key in keyof TObject]?: UnwrapSecured<TObject[Key]> };
     nodevar: string;
   }) {
     return await this.updateProperties({
       type: upperFirst(nodevar),
       object,
-      props,
       changes,
     });
   }
 
+  // expecting the changes of the "simple" properties.
+  async getActualChanges<TResource extends ResourceShape<any>>(
+    oldObject: TResource['prototype'],
+    changes: Partial<Record<keyof TResource['prototype'] & string, any>>
+  ): Promise<typeof changes> {
```

Comment:
Does this typeof work? Maybe it would be better to define a `ChangesOf<Resource>` type alias.

Reply:
It has been working, but I'm fine with making an alias. Honestly though, aliases kind of take away readability for me personally.